### PR TITLE
Lightweight dApp Development Guide

### LINKED ISSUE #7300: Create a new Guide about Building Light Weight dApps

### PR DESCRIPTION
**Pull request by danforbes**
Closes #7300

### CODE REVIEW

File: docs/docs/guides/07_dapps/lightweight-dapp.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+Here is a high-level overview of the steps in this tutorial:
+
+1. Review prerequisites
+2. Initialize a new Web3.js dApp and add dependencies
+3. Configure and analyze a development build
+4. Analyze the built-in React production build
+5. Use only necessary packages and analyze the result
+
+:::tip
+If you encounter any issues while following this guide or have any questions, don't hesitate to seek assistance. Our friendly community is ready to help you out! Join our [Discord](https://discord.gg/F4NUfaCC) server and head to the **#web3js-general** channel to connect with other developers and get the support you need.
```
Comment (jdevcs) [inline docs/docs/guides/07_dapps/lightweight-dapp.md]: https://discord.gg/F4NUfaCC is not valid.

File: docs/docs/guides/07_dapps/lightweight-dapp.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+10.8.2
+```
+
+## Step 2: Initialize a New Web3.js dApp and Add Dependencies
+
+Use `create-web3js-dapp` to initialize a new Web3.js dApp and navigate into the new project directory:
+
+```console
+npx create-web3js-dapp --framework react --template minimal
+cd web3js-react-dapp-min
```
Comment (jdevcs) [inline docs/docs/guides/07_dapps/lightweight-dapp.md:47]:
```suggestion
npm install -g create-web3js-dapp
npx create-web3js-dapp --framework react --template minimal
cd web3js-react-dapp-min
```

File: docs/docs/guides/07_dapps/lightweight-dapp.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+```
+
+At the time of writing this guide, the size of the bundle generated by the built-in production build is about **593 kilobytes**, which is about 2% the size of the initial development build. The reduction in bundle size is due to techniques such as tree shaking. After reviewing the output of Webpack Bundle Analyzer, close the browser tab and use `Ctrl + C` to cancel the Webpack process.
+
+## Step 5: Use Only Necessary Packages and Analyze the Result
+
+The minimal template dApp that was created by `create-web3js-dapp` uses the umbrella `web3` package, which is useful for development purposes but results in a relatively large bundle size. In fact, the capabilities of the minimal React template only require the `web3-eth` and `web3-types` packages. To replace the umbrella `web3` package with `web3-eth` and `web3-types`, execute the following commands:
+
+```console
+npm remove web3
```
Comment (jdevcs) [inline docs/docs/guides/07_dapps/lightweight-dapp.md]:
```suggestion
npm uninstall web3
```

File: docs/docs/guides/07_dapps/lightweight-dapp.md (hunk 1 above)
Comment (danforbes) [reply docs/docs/guides/07_dapps/lightweight-dapp.md]: Good catch 🙏🏻 I've updated this in all the guides/tutorials.

**Review by avkos — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

File: docs/docs/guides/07_dapps/lightweight-dapp.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+10.8.2
+```
+
+## Step 2: Initialize a New Web3.js dApp and Add Dependencies
+
+Use `create-web3js-dapp` to initialize a new Web3.js dApp and navigate into the new project directory:
+
+```console
+npm install -g create-web3js-dapp
+npx create-web3js-dapp --framework react --template minimal
```
Comment (krzysu) [inline docs/docs/guides/07_dapps/lightweight-dapp.md:46]: I thought the whole idea of npx is that there is no need to install a package globally

**Pull request merged**